packages/framework/composition/src/composeFactory.ts

```diff
@@ -75,7 +77,7 @@ export function composeFactory<TProps, TSlotProps, TTokens, TTheme, TStatics ext
       themeHelper,
     );
 
-  component.compose = (customOptions: Partial<LocalOptions>) =>
+  component.compose = (customOptions: TwoLevelPartial<LocalOptions>) =>
     composeFactory<TProps, TSlotProps, TTokens, TTheme, TStatics>(
       immutableMergeCore(mergeOptions, options, customOptions) as LocalOptions,
```

Comment:
I wonder if this actually should be a DeepPartial, aka, every type all the way down is optional. The reason being, the call to `immutableMergeCore` will just merge anything we provide on top of the existing composed component. So in effect, every option is optional. Thoughts?

Reply:
For reference, how to implement a DeepPartial: https://stackoverflow.com/a/61132308